enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,1016 @@
+```json
+{
+  "name": "tenantred",
+  "namespace": "default",
+  "interface": "pod16367aacb67",
+  "cni-args": {
+    "ipamclaim.cni.cncf.io/ownerID": "a0790345-4e84-4257-837a-e3d762d191ab",
+    "ipamclaim.cni.cncf.io/ownerName": "vm-a",
+    "ipamclaim.cni.cncf.io/ownerType": "VirtualMachine",
+    "ipamclaim.cni.cncf.io/ownerVersion": "kubevirt.io/v1"
+  }
+}
+```
+
+When the CNI plugin notices this information in the network selection element,
+it allocates an IP address from the pool, and then creates the CR with the IPs
+directly in the spec - keep in mind there's no reason for this alternative to
+feature the status sub-resource.
+
+The CNI plugin will also create the `IPAMClaim` with the owner reference
+information, which was passed via CNI-Args. The CR would look like:
+```yaml
+apiVersion: "ipamclaims.k8s.cni.cncf.io/v1alpha1"
+kind: IPAMClaim
+metadata:
+  name: vm-a.tenantred-attachment
+  namespace: ns1
+  ownerReferences:
+  - apiVersion: kubevirt.io/v1
+    kind: VirtualMachine
+    name: vm-a
+    uid: a0790345-4e84-4257-837a-e3d762d191ab
+spec:
+  network: tenantred
+  interface: pod16367aacb67
+  ips:
+  - 10.128.20.8/24
+  - fd10:128:20::8/64
+```
+
+This alternative cannot handle the scenario where foreground deletion is asked,
+since OVN-Kubernetes (the component setting the finalizer) does not know when
+to remove the finalizer (i.e. it doesn't know the VMI lifecycle).
+
+This alternative also requires KubeVirt to template the pod indicating the name
+of the `IPAMClaim`, as described in the [pod templating](#pod-templating)
+section.
+
+### Delegate the IPAM functionality to a CNI IPAM plugin
+
+Below you can find a Network Attachment Definition requesting delegated IPAM
+from a separate IPAM plugin.
+
+```yaml
+---
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: whereabouts-conf
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "l2-network",
+      "type": "ovn-k8s-cni-overlay",
+      "topology":"layer2",
+      "mtu": 1300,
+      "netAttachDefName": "default/whereabouts-conf",
+      "ipam": {
+        "type": "whereabouts",
+        "log_file": "/var/log/wb.log",
+        "log_level": "debug",
+        "range": "192.168.2.0/24"
+      }
+    }'
+```
+
+Thus, we would have the following three behaviors:
+- cni conf without a subnet defined **and** without IPAM key defined => only
+  static IP can be requested
+- cni conf without a subnet defined **but with** the IPAM key defined =>
+  delegate IPAM to an external plugin
+- cni conf with the subnet key defined => OVN-K provided IPAM for the provided
+  subnet
+
+**NOTE**: it results in an error if the user attempts to define **both** the
+subnet **and** the IPAM key at the same time.
+
+#### Why this alternative was not considered further
+
+Implementing the sticky IPs feature in separate IPAM plugins will prevent this
+solution from being used on the default cluster network for the HyperShift
```

Comment:
Supporting the default cluster network is non-goal, so not sure how strong this argument is